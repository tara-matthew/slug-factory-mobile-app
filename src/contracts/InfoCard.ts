export interface IInfoCardProps {
    imageUrl: string;
    name: string;
    uploadCount: number;
    info: Array<string>;
}
