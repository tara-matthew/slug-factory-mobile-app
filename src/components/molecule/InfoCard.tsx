import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import ContentWithDivider from "./ContentWithDivider";
import { DividerWidth } from "../../contracts/Divider";
import { IInfoCardProps } from "../../contracts/InfoCard";

const InfoCard = ({ name, uploadText, imageUrl, info }: IInfoCardProps) => {
    return (
        <View style={ styles.container } className="rounded-lg shadow-sm p-8">
            <View className="flex flex-row items-center">
                <View className="justify-center items-center">
                    <Image
                        style={ {
                            width: 75,
                            height: 75,
                            borderRadius: 75,
                            marginBottom: 8,
                        } }
                        resizeMode="cover"
                        source={ {
                            uri: imageUrl,
                        } }
                    />

                    <Text
                        className="text-xl font-bold text-center"
                        style={ { maxWidth: 120, textAlign: "center" } }
                        numberOfLines={ 1 }
                    >
                        {name}
                    </Text>

                    <Text
                        className="text-lg text-center"
                        style={ { maxWidth: 120, textAlign: "center" } }
                    >
                        {uploadText}
                    </Text>
                </View>

                <View className="flex-1 flex flex-col">
                    {info.map((item, index) => {
                        const isLast = index === info.length - 1;

                        return (
                            <ContentWithDivider
                                outerClass="items-end"
                                key={ index }
                                top={ false }
                                bottom={ !isLast }
                                dividerWidth={ DividerWidth.Half }
                            >
                                <Text
                                    numberOfLines={ 1 }
                                    style={ { maxWidth: 160 } }
                                    className="text-lg py-2"
                                >
                                    {item}
                                </Text>

                            </ContentWithDivider>
                        );
                    })}
                </View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#d0cadb", // '#e8e8e8', //"#e8eaeb",
    },
    divider: {
        width: "50%",
        height: 1,
        backgroundColor: "gray",
    },
});

export default InfoCard;
