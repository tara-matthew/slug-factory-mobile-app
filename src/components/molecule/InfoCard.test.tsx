import React from "react";
import InfoCard from "./InfoCard";
import { render, fireEvent, screen } from "@testing-library/react-native";

it("renders correctly", () => {
    const { toJSON } = render(<InfoCard imageUrl="/prints/1" info={["some info", "some more info"]} name="Test name" uploadText="Upload text"/>);

    expect(toJSON()).toMatchSnapshot();
});

it.todo("displays an image");
it.only("does not render a divider on the bottom of the last info item", () => {
    const {getByText} = render(<InfoCard imageUrl="/prints/1" info={["some info", "other info", "some more info"]} name="Test name" uploadText="Upload text"/>)
    const lastItem = getByText('some more info');
    const firstItem = getByText("some info");

    let lastItemParent = lastItem.parent;

    while (lastItemParent && lastItemParent.type?.name !== 'ContentWithDivider') {
        lastItemParent = lastItemParent.parent;
    }

    let firstItemParent = firstItem.parent;

    while (firstItemParent && firstItemParent.type?.name !== 'ContentWithDivider') {
        firstItemParent = firstItemParent.parent;
    }

    // check that the last item does not have a divider on the bottom
    expect(lastItemParent).not.toBeNull();
    expect(lastItemParent.props.bottom).toBe(false);
    // check that an item which isn't the last one does have a divider on the bottom
    expect(firstItemParent).not.toBeNull();
    expect(firstItemParent.props.bottom).toBe(true);

});
