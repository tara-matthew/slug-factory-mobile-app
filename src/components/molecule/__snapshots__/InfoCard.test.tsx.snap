// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="rounded-lg shadow-sm p-8"
  style={
    {
      "backgroundColor": "#d0cadb",
    }
  }
>
  <View
    className="flex flex-row items-center"
  >
    <View
      className="justify-center items-center"
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": "/prints/1",
          }
        }
        style={
          {
            "borderRadius": 75,
            "height": 75,
            "marginBottom": 8,
            "width": 75,
          }
        }
      />
      <Text
        className="text-xl font-bold text-center"
        numberOfLines={1}
        style={
          {
            "maxWidth": 120,
            "textAlign": "center",
          }
        }
      >
        Test name
      </Text>
      <Text
        className="text-lg text-center"
        style={
          {
            "maxWidth": 120,
            "textAlign": "center",
          }
        }
      >
        Upload text
      </Text>
    </View>
    <View
      className="flex-1 flex flex-col"
    >
      <View
        className="items-end"
      >
        <View>
          <Text
            className="text-lg py-2"
            numberOfLines={1}
            style={
              {
                "maxWidth": 160,
              }
            }
          >
            some info
          </Text>
        </View>
        <View
          className="w-1/2"
          style={
            {
              "backgroundColor": "gray",
              "height": 1,
            }
          }
        />
      </View>
      <View
        className="items-end"
      >
        <View>
          <Text
            className="text-lg py-2"
            numberOfLines={1}
            style={
              {
                "maxWidth": 160,
              }
            }
          >
            some more info
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
